src/CRISP.BackendChallenge/Controllers/EmployeeController.cs

```diff
@@ -7,33 +7,266 @@
 namespace CRISP.BackendChallenge.Controllers;
 
 [ApiController]
-[Route("[controller]")]
+[Route("api/[controller]")]
 public class EmployeeController : ControllerBase
 {
     private readonly ILogger<EmployeeController> _logger;
-    private readonly IRepository<Employee> _repository;
+    private readonly IRepository<Employee> _employeeRepository;
+    private readonly IRepository<Login> _loginRepository;
 
-    public EmployeeController(ILogger<EmployeeController> logger, IRepository<Employee> repository)
+    public EmployeeController(ILogger<EmployeeController> logger, IRepository<Employee> employeeRepository, IRepository<Login> loginRepository)
     {
         _logger = logger;
-        _repository = repository;
+        _employeeRepository = employeeRepository;
+        _loginRepository = loginRepository;
     }
 
     [HttpGet]
+    [Route("all")]
     public IActionResult GetAll()
     {
         _logger.LogDebug(":: Performing {MethodName}", nameof(GetAll));
-        var result = _repository.Query()
-            .ToList().Select(x => new EmployeeResponse
+            
+        try
+        {
+            var employees = _employeeRepository.GetAll();
+
+            var result = employees.Select(ToEmployeeResponse);
+
+            return Ok(result);
+        }
+        catch (ArgumentNullException e)
+        {
+            _logger.LogDebug(":: There are no Employees in the Database: {EMessage}", e.Message);
+            return NotFound(new {error = "No Employees Found in Database"});
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(":: Error occurred while getting all employees: {EMessage}", e.Message);
+            return CreateInternalServerError();
```

Comment:
All of the endpoints have a very similar line. What would be preferable is am exception handling middleware at the controller level, or to not expect to need to handle arbitrary exceptions in your controller at all

Reply:
Created a exception filter middleware for the controller level for a catch all stray exceptions